crates/networking/rpc/engine/fork_choice.rs

```diff
@@ -75,6 +75,35 @@ impl RpcHandler for ForkChoiceUpdatedV2 {
     }
 }
 
+fn validate_attributes_v1_v2(
+    attributes: &PayloadAttributesV3,
+    head_block: &BlockHeader,
+    context: &RpcApiContext,
+) -> Result<(), RpcErr> {
+    let chain_config = context.storage.get_chain_config()?;
+    if attributes.parent_beacon_block_root.is_some() {
+        return Err(RpcErr::InvalidPayloadAttributes(
+            "Attribute parent_beacon_block_root is non-null".to_string(),
+        ));
+    }
+
+    match chain_config.get_fork(attributes.timestamp) {
```

Comment:
Minor detail, but maybe I'd split this validate function into v1 and v2 to avoid this match just like for execution_payload.